### DISCUSSION TITLE
A question on ECOD and COPOD

### DISCUSSION THREAD

**Discussion by bl3e967** [yzhao062/pyod#548] [General]
Hello, 

I have a question on the implementation of COPOD and its relationship with ECOD. 

Having looked through the paper for COPOD, it seems that the methodology used for this model makes an independence assumption between each dimension of a dataset in order to compute the neg-log-likelihood as the sum of the neg-log-likelihood of each marginal. 

This assumption means that in essence, we are not using a copula at all - which I think is why the paper on ECOD was written to acknowledge this fact and better address the implications of this assumption.

Assuming I am correct on the above, I'm unsure why we are maintaining both COPOD and ECOD in this package. Shouldn't COPOD be deprecated in favour of ECOD? I'm happy to be corrected if I am wrong, which is why I write this in the discussions section and not as an issue. 

@yzhao062

**Comment by Lucew** [comment]
There might be two accompanying issues #453 and #561 
I'm also interested.

**Comment by samschoo** [comment]
Hi,

Last year I was investigating COPOD and I also noticed that the dependency between dimensions was not addressed, given the exclusive use of marginal ECDFs (it should noted that I'm not very knowledgeable on copulas). Upon seeing some agreement here, I read both COPOD and ECOD papers again and there are numerous similarities between the papers. I've concluded that these algorithms are exactly the same. It's also odd that the authors (the same authors for ECOD, one of which created pyod), make no mention of COPOD in ECOD, but maybe that's an academic formality I'm unaware of.

Needless to say, the existence of COPOD along side ECOD (apart from their incorrect implementation #493) is misleading so I'm going to create an issue on this. (#655)

Thanks,
Sam
